### PR TITLE
[attributesprocessor] Support filter by severity

### PR DESCRIPTION
**Pull request by atoulme**
**Description:** 
Adds a severity text matcher for the attributesprocessor.

**Link to tracking Issue:**
No issue.

**Testing:**
Unit tests

**Documentation:** 
README and configuration

### CODE REVIEW

File: internal/coreinternal/processor/filterlog/filterlog.go
Comment (djaglowski) [inline internal/coreinternal/processor/filterlog/filterlog.go:106]: Log "severity" is technically a number, whereas "severity text" is a string. 

I think we can anticipate that filtering will be necessary on numeric severity as well. Would that fit into this configuration option or be separate. If separate, should this one be called something else to avoid the future naming collision?

**Comment by pmm-sumo** [comment]
I think the change looks good overall. I think before merging it would be good taking the item raised by @djaglowski, i.e. if we want to have separate filter for log severity number and if that's the case then how the configuration keys should be named for them.

As a digression, I think that while having ability to filter records when attributes processor is concerned is helpful, it would be even more useful to have similiar capabilities exposed in [filterprocessor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/filterprocessor) as well

**Comment by djaglowski** [comment]
> As a digression, I think that while having ability to filter records when attributes processor is concerned is helpful, it would be even more useful to have similiar capabilities exposed in [filterprocessor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/filterprocessor) as well

+1, This processor is is filtering on several non-attributes at this point.

File: internal/coreinternal/processor/filterlog/filterlog.go
Comment (atoulme) [reply internal/coreinternal/processor/filterlog/filterlog.go]: OK addressed.

**Review by codeboten — APPROVED** [review]
The change looks good to me. @djaglowski @pmm-sumo please review and approve if your comments have been addressed.

**Comment by pmm-sumo** [comment]
> @pmm-sumo would you be awesome and file an issue for the digression? That sounds like a good move.

Sure thing: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/9235

> The problem with severity number fwiw is that you'd be well tempted to use a `>`, `<`, `>=` or `<=` to filter by, and that seemed like something just a bit more complex.

I think that filtering by severity text is already pretty useful, don't want to block this PR due to lack of filtering by number but I wanted to hear from @djaglowski on his take on that :)

**Review by pmm-sumo — APPROVED** [review]

**Review by djaglowski — APPROVED** [review]
Sorry for the delay. LGTM

**Comment by TylerHelmuth** [comment]
Tangentially related, when should new capabilities be added to the attribute processor vs the transform processor?

**Comment by pmm-sumo** [comment]
> Tangentially related, when should new capabilities be added to the attribute processor vs the transform processor?

That's a good question. In my opinion it's OK to add small features here that users want and then port those to transform processor after it's considered ready to be used

There is a [PR](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/9368#pullrequestreview-949005483) that adds some log processing capabilities. This makes it already quite useful IMO

**Pull request merged**
